nemo/collections/nlp/models/language_modeling/megatron_base_model.py

```diff
@@ -765,16 +768,16 @@ def build_model_parallel_config(self) -> ModelParallelConfig:
         config_mapping = {
             "perform_initialization": True,  # initailize weights when constructing the module
             "fp16": False,  # NeMo does not currently support fp16 training with megatron amp O2
-            "bf16": precision == 'bf16' and megatron_amp_O2,
+            "bf16": self.torch_dtype == torch.bfloat16 and megatron_amp_O2,
             "params_dtype": params_dtype,
             "timers": None,  # NeMo does not currently support megatron core timers
             "async_tensor_model_parallel_allreduce": self.cfg.get('tensor_model_parallel_world_size', 1) > 1
             and not self.cfg.get('sequence_parallel', False),
             "pipeline_dtype": pipeline_dtype,
             "grad_scale_func": self.trainer.precision_plugin.scaler.scale
-            if precision in [16, '16', '16-mixed']
+            if self.torch_dtype == torch.float16
```

Comment:
Just noting here that for inference or eval we won't need the grad scale function and it might not exist. We might want to update the behavior based on whether we're training or not.